velox/core/QueryCtx.h

```diff
@@ -28,6 +28,15 @@ namespace facebook::velox::core {
 
 class QueryCtx {
  public:
+  /// Threading mode the query uses to execute. 'kSerial' mode executes the
```

Comment:
Can we just define Mode as sub class of Task and pass this to Task on Task::create?